### ISSUE TITLE
Milestone 2 Code Feedback

### ISSUE THREAD

**Issue by ahofmann4** [UBC-MDS/CPD#57]
General code quality comments:
there is some repetition in the eda file to make graphs, particularly histograms maybe a recursive function could be used instead. 
Tests not present in functions.

When I went to run your code I ran into a few problems, see below:
---------------------- Errors from running codes, all caused by Rscript instruction being wrong
- one code does not work: Rscript src/linear_regression_analysis --file_path=data/processed --out_dir=results
Fatal error: cannot open file 'src/linear_regression_analysis': No such file or directory
-file name needs .R
- still get errors, can't find the source.. 

-----------------------
Calls: main ... standardise_path -> detect_compression -> readBin -> file
In addition: Warning messages:
1: In file(con, "rb") :
  'raw = FALSE' but '/Users/analisehofmann/Desktop/M2_Marking/CPD-0.1.0/data/processed' is not a regular file
2: In file(con, "rb") :
  cannot open file '/Users/analisehofmann/Desktop/M2_Marking/CPD-0.1.0/data/processed': it is a directory
Execution halted

--file_path=data/processed must be --file_path=data/processed/complaints_police.csv to make the command work

------------------

**Comment by eboylestanley** [comment]
@ahofmann4 Thank you for the feedback!  What functions in particular would need a test?  We weren't sure what types of test would be needed in our scripts.

**Comment by ahofmann4** [comment]
From what I understand, the expectations of the program is that anytime you
write your own function you should ideally have a test at the end to check
that your outputs are correct.

All packages you download and use have these tests/checks built into them
already so you don't have to check outputs of package calls you are making.

I believe you have learned in another course this block about tests, so
those guidelines hold. If you have further questions regarding tests,
please post them in the slack. I'm sure other groups have a similar
question, and then Tiff can provide further clarification or point you to
the appropriate MDS materials.

**Comment by elabandari** [comment]
Hi @ttimbers, 

Could you please give us some concrete suggestion of tests to include?

**Comment by ttimbers** [comment]
Just to clarify, main functions do not need tests, but other functions do. Also, because we have not trained you how to test visualization/plot objects in MDS yet, you do not have to write tests for such functions either.

**Comment by elabandari** [comment]
Hi Analise, 

Thank you for your feedback. It was very helpful as we were finalizing our project. 

Below is how we have addressed your feedback: 

There is some repetition in the eda file to make graphs, particularly histograms maybe a recursive function could be used instead.
- We have made a function for plotting the histograms. 

Tests not present in functions.
- We have tried to program defensively but since no function other than the main function were present no other tests were required.

Issues running code 
- Corrected makefile and tested that docker and makefile run on multiple devices.

Thanks again!
